### DISCUSSION TITLE
order by during aggregation

### DISCUSSION THREAD

**Discussion by ac4wong** [duckdb/duckdb#4864] [Q&A] [answered]
https://duckdb.org/docs/sql/aggregates
"When the ORDER BY clause is provided, the values being aggregated are sorted before applying the function. Usually this is not important, but there are some order-sensitive aggregates that can have indeterminate results (e.g., first, last, list and string_agg)."

This is exactly what I want to do but I cannot figure out the syntax.  Can you please provide an example?
e.g. I want to first sort the row by "date" and pick the last value of each month.

**Comment by Alex-Monahan** [comment]
Howdy! Maybe something like this?
```sql
Select 
  month_col,
  last(value_col) as last_value
From my_table
Group by
  Month_col
Order by 
  date asc
```

**Reply by ac4wong** [reply]
so I did something like this

_select datetrunc('month', date), last (date), last(factorValue) from
	mytable
WHERE 
factorname in ('FactorMktcap1')
and barraid = 'JPNGME1'
group by datetrunc('month', date)
order by date_

This gives me "SQL Error: java.sql.SQLException: Binder Error: column "date" must appear in the GROUP BY clause or be used in an aggregate function"

If I remove the "order by date" at the end, it will run but obviously it doesn't do what I want since the order is messed up.

**Comment by Mytherin** [comment, marked as answer]
You can put an `ORDER BY` clause after the aggregate parameters, e.g.:

```sql
SELECT FIRST(id ORDER BY id DESC), FIRST(id ORDER BY  id ASC) FROM range(10) tbl(id);
┌────────────────────────────┬───────────────────────────┐
│ first(id ORDER BY id DESC) │ first(id ORDER BY id ASC) │
├────────────────────────────┼───────────────────────────┤
│ 9                          │ 0                         │
└────────────────────────────┴───────────────────────────┘
```

**Reply by Mytherin** [reply]
I have [added an example for this to the aggregate page](https://duckdb.org/docs/sql/aggregates)
